### PR TITLE
Dispose logger factory when disposing hosting engine

### PR DESCRIPTION
**Pull request by kichalla**
@Tratcher This is related to the PR https://github.com/aspnet/Logging/pull/162

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs [hunk 1]
```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```
Comment (kichalla) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: It feels odd to just dispose this service here. Not sure if there are any other services which need similar requirement.

Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: Agreed, this is weird. Isn't this what DI scopes are for? Would it make sense to create a scope on line 66 and dispose it here? Then any scoped services would be disposed.

Comment (kichalla) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: Hmm...I have seen DI scopes used in `per-request` kind of scenarios where as logger factory here is an application wide service. (thinking of it in a different way...the application start and shutdown itself is a scope and probably we should just dispose off all disposable application services as how DI scope would do).

Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: Consider in the test scenario where we start multiple TestHost/Servers in one process.  HostingEngine is only a scoped subset of an actual application, so I think DI scopes are an appropriate concept here.  Per-request scopes would be nested inside of the per-HostingEngine scopes.

Comment (kichalla) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: :+1:

**Comment by kichalla** [comment]
@rynowak

File: src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs (hunk 1 above)
Comment (kichalla) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: @HaoK  and I discussed about this and agreed that disposing loggerfactory alone for this PR is fine and we will file another issue to handle disposing all the hosting engine services as it requires more refactoring than expected.

@Tratcher

Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: Why bother with the partial fix then? It's not urgent.

Comment (kichalla) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: The following PR depends on this
https://github.com/aspnet/Logging/pull/162

Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs]: No, Hosting depends on Logging. That Logging PR is nice to have, but calling Dispose not a blocking issue for beta7.

Your partial fix here does not contribute towards the full fix, so don't bother.

**Comment by kichalla** [comment]
Closing this PR as it will be handled as part of fix for https://github.com/aspnet/Hosting/issues/322

**Pull request closed without merging**